deploy/kubelessDeploy.js

```diff
@@ -12,46 +12,50 @@ class KubelessDeploy {
 
     this.hooks = {
       'deploy:deploy': () => BbPromise.bind(this)
-        .then(this.deployFunction)
+        .then(this.validate)
+        .then(this.deployFunction),
     };
   }
 
-  deployFunction() {
-    this.serverless.cli.log(`Deploying function: ${this.serverless.service.service}...`);
+  validate() {
+    helpers.validateEnv();
+    return BbPromise.resolve();
+  }
 
-    var funcs = {
-      apiVersion: `k8s.io/v1`,
+  deployFunction() {
+    const f = this.serverless.service.service;
+    this.serverless.cli.log(`Deploying function: ${f}...`);
+    const funcs = {
+      apiVersion: 'k8s.io/v1',
       kind: 'Function',
       metadata: {
         name: this.serverless.service.service,
         namespace: 'default',
       },
       spec: {
         deps: '',
-        function: fs.readFileSync(this.serverless.service.functions.hello.handler.toString().split(".")[0] + ".py", 'utf8'),
-        handler: this.serverless.service.functions.hello.handler,
+        function: this.serverless.utils.readFileSync(
+          `${this.serverless.service.functions[f].handler.toString().split('.')[0]}.py`
+        ),
+        handler: this.serverless.service.functions[f].handler,
         runtime: this.serverless.service.provider.runtime,
         topic: '',
         type: 'HTTP',
       },
     };
 
-    var thirdPartyResources = new Api.ThirdPartyResources({
-      url: process.env['K8SAPISERVER'],
-      ca: fs.readFileSync(process.env['HOME'] + "/.minikube/ca.crt"),
-      cert: fs.readFileSync(process.env['HOME'] + "/.minikube/apiserver.crt"),
-      key: fs.readFileSync(process.env['HOME'] + "/.minikube/apiserver.key"),
-      group: 'k8s.io',
-    });
+    const thirdPartyResources = new Api.ThirdPartyResources(
+      Object.assign(helpers.getMinikubeCredentials(), {
```

Comment:
let's open an issue to track the case where the k8s endpoint is not minikube but on GKE or some manually setup `kubeadm` cluster. We need to make sure we can get the creds properly. We can do this in a separate PR.